firefox_ui_tests/remote/security/test_safe_browsing_warning_pages.py

```diff
@@ -68,31 +68,28 @@ def check_get_me_out_of_here_button(self, unsafe_page):
         self.wait_for_condition(lambda mn: self.browser.default_homepage in mn.get_url())
 
     def check_report_button(self, unsafe_page):
+        # Get the URL of the support site for phishing and malware. This may result in a redirect.
+        with self.marionette.using_context('chrome'):
+            url = self.marionette.execute_script("""
+              Components.utils.import("resource://gre/modules/Services.jsm");
+              return Services.urlFormatter.formatURLPref("app.support.baseURL")
+                                                         + "phishing-malware";
+            """)
+
         button = self.marionette.find_element(By.ID, "reportButton")
         button.click()
 
         # Wait for the button to become stale, then wait for page load
-        # so we can verify the url
+        # so we can verify the url even if a redirect happens
         self.wait_for_condition(expected.element_stale(button))
 
-        # Get the base URL to check; this will result in a redirect.
-        with self.marionette.using_context('chrome'):
-            if 'its-a-trap' in unsafe_page:
-                url = self.marionette.execute_script("""
-                  Components.utils.import("resource://gre/modules/Services.jsm");
-                  return Services.urlFormatter.formatURLPref("app.support.baseURL")
-                                                             + "phishing-malware";
-                """)
-            else:
-                url = self.marionette.execute_script("""
-                  Components.utils.import("resource://gre/modules/Services.jsm");
-                  return Services.urlFormatter.formatURLPref(
-                  "browser.safebrowsing.malware.reportURL") + arguments[0];
-                """, script_args=[unsafe_page])
+        # TODO: Bug 1140470: use replacement for mozmill's waitforPageLoad
+        self.wait_for_condition(lambda mn: mn.execute_script("""
+          return document.readyState == 'complete';
+        """))
```

Comment:
I don't see how this is achieving something different from waiting for the urls to be the same, but either way this reflects the browser change and is still effective.